TestModelServerController_PodLifecycle  failure

--- FAIL: TestModelServerController_PodLifecycle (10.13s)
    --- FAIL: TestModelServerController_PodLifecycle/PodCreateReady (2.00s)
        modelserver_controller_test.go:934: Object not found in cache after 2s timeout
        modelserver_controller_test.go:348: 
            	Error Trace:	/home/runner/work/matrixinfer/matrixinfer/pkg/infer-gateway/controller/modelserver_controller_test.go:348
            	Error:      	Should be true
            	Test:       	TestModelServerController_PodLifecycle/PodCreateReady
            	Messages:   	Pod should be found in store after creation
    --- FAIL: TestModelServerController_PodLifecycle/PodCreateNotReady (2.01s)
        modelserver_controller_test.go:934: Object not found in cache after 2s timeout
        modelserver_controller_test.go:395: 
            	Error Trace:	/home/runner/work/matrixinfer/matrixinfer/pkg/infer-gateway/controller/modelserver_controller_test.go:395
            	Error:      	Should be true
            	Test:       	TestModelServerController_PodLifecycle/PodCreateNotReady
            	Messages:   	Pod should be found in store after creation
    --- FAIL: TestModelServerController_PodLifecycle/PodUpdateBecomesReady (2.00s)
        modelserver_controller_test.go:934: Object not found in cache after 2s timeout
        modelserver_controller_test.go:440: 
            	Error Trace:	/home/runner/work/matrixinfer/matrixinfer/pkg/infer-gateway/controller/modelserver_controller_test.go:440
            	Error:      	Should be true
            	Test:       	TestModelServerController_PodLifecycle/PodUpdateBecomesReady
            	Messages:   	Pod should be found in store after update
    --- FAIL: TestModelServerController_PodLifecycle/PodUpdateBecomesNotReady (4.00s)
        modelserver_controller_test.go:934: Object not found in cache after 2s timeout
        modelserver_controller_test.go:4[74](https://github.com/matrixinfer-ai/matrixinfer/actions/runs/16747145729/job/47408301549?pr=232#step:4:75): 
            	Error Trace:	/home/runner/work/matrixinfer/matrixinfer/pkg/infer-gateway/controller/modelserver_controller_test.go:474
            	Error:      	Should be true
            	Test:       	TestModelServerController_PodLifecycle/PodUpdateBecomesNotReady
            	Messages:   	Pod should be found in store after creation
        modelserver_controller_test.go:934: Object not found in cache after 2s timeout
        modelserver_controller_test.go:490: 
            	Error Trace:	/home/runner/work/matrixinfer/matrixinfer/pkg/infer-gateway/controller/modelserver_controller_test.go:490
            	Error:      	Should be true
            	Test:       	TestModelServerController_PodLifecycle/PodUpdateBecomesNotReady
            	Messages:   	Pod should not be found in store after creation

--- a/pkg/infer-gateway/controller/modelserver_controller_test.go
+++ b/pkg/infer-gateway/controller/modelserver_controller_test.go
@@ -334,13 +334,6 @@ func TestModelServerController_PodLifecycle(t *testing.T) {
 			context.Background(), pod, metav1.CreateOptions{})
 		assert.NoError(t, err)
 
-		// Clear queue before test
-		for controller.workqueue.Len() > 0 {
-			item, _ := controller.workqueue.Get()
-			controller.workqueue.Done(item)
-			controller.workqueue.Forget(item)
-		}
-
 		sync := waitForObjectInCache(t, 2*time.Second, func() bool {
 			pods, _ := store.GetPodsByModelServer(utils.GetNamespaceName(ms))
 			return len(pods) > 0 && pods[0].Pod.Name == "test-pod-ready"
@@ -374,12 +367,6 @@ func TestModelServerController_PodLifecycle(t *testing.T) {
 			context.Background(), pod, metav1.CreateOptions{})
 		assert.NoError(t, err)
 
-		// Clear queue before test
-		for controller.workqueue.Len() > 0 {
-			item, _ := controller.workqueue.Get()
-			controller.workqueue.Done(item)
-			controller.workqueue.Forget(item)
-		}
 		sync := waitForObjectInCache(t, 2*time.Second, func() bool {
 			pods, _ := controller.podLister.Pods("default").Get("test-pod-not-ready")
 			return pods != nil && pods.Name == "test-pod-not-ready"

--- a/pkg/infer-gateway/datastore/model_server.go
+++ b/pkg/infer-gateway/datastore/model_server.go
@@ -45,9 +45,10 @@ func newModelServer(ms *aiv1alpha1.ModelServer) *modelServer {
 }
 
 func (m *modelServer) getPods() []types.NamespacedName {
-	podNames := make([]types.NamespacedName, 0, m.pods.Len())
 	m.mutex.RLock()
 	defer m.mutex.RUnlock()
+	podNames := make([]types.NamespacedName, 0, m.pods.Len())
+
 	for podName := range m.pods {
 		podNames = append(podNames, podName)
 	}

